graphql/rules.lua

```diff
@@ -159,15 +158,17 @@ function rules.unambiguousSelections(node, context)
 
         validateField(key, fieldEntry)
       elseif selection.kind == 'inlineFragment' then
-        local parentType = selection.typeCondition and context.schema:getType(selection.typeCondition.name.value) or parentType
-        validateSelectionSet(selection.selectionSet, parentType)
+        local _parentType = selection.typeCondition
+          and context.schema:getType(selection.typeCondition.name.value)
+          or parentType
+        validateSelectionSet(selection.selectionSet, _parentType)
```

Comment:
And here too.

Reply:
Same here. Would you suggest more appropriate change? Without refactoring. This project has not so much tests yet

Reply:
I don't think that the warning about redefinition of a variable should be enabled by default: too much false positives (I mean, code where it is not a problem) and artificial changes like this one.

Reply:
Tried to stick to the luacheck [tarantool standard](https://github.com/tarantool/luacheck/pull/8) here

Reply:
Ugh. It seems that the decision was made, so redefinitions are banned. Okay.

Let's choose a meaningful name so. The value is a type from `on` clause if it exists. We can name it either `onConditionType` or `nextParentType`: one way is how it is obtained, another how it intended to be used. I think that the former way improves readability more then the latter.